src/main/java/org/jfrog/gradle/plugin/artifactory/extractor/GradleModuleExtractor.java

```diff
@@ -99,6 +102,21 @@ private ModuleBuilder getModuleBuilder(Project project, Set<GradleDeployDetails>
         return builder;
     }
 
+    /**
+     * Get the module type to publish.
+     *
+     * @param project - Project to extract the module type
+     * @return the module type.
+     */
+    private ModuleType getModuleType(Project project) {
+        PluginContainer plugins = project.getPlugins();
+        if (!plugins.withType(CppApplicationPlugin.class).isEmpty() ||
+                !plugins.withType(CppLibraryPlugin.class).isEmpty()) {
```

Comment:
Shouldn't this be an AND condition instead of OR?

Reply:
No. We need **one of** the plugin lists to be not empty.
In other words -
The `cpp-application` plugin **or** the `cpp-library` plugin should be applied.

Reply:
Add the following import statement, and remove the ones related to the CppApplication and CppLibrary plugins
```
import org.gradle.language.plugins.NativeBasePlugin;
```

Then add:

```
     private ModuleType getModuleType(Project project) {
         PluginContainer plugins = project.getPlugins();
         if (!plugins.withType(NativeBasePlugin.class).isEmpty()) {
             return ModuleType.CPP;
         }
         return ModuleType.GRADLE;
     }
```

Sorry it took me so long to get that tested. I had to fix up the build to publish to mavenLocal and then re-run my separate build using that. In addition, the ModuleType.CPP change wasn't in the build-info-api yet so I had to additionally publish that locally too.

But the above works, it fills in the module with a type="cpp" field, and any dependencies under it seem to inherit it (despite being empty in the build-info.json file) when published to artifactory. They do show up as CPP type for X-Ray / Build SBOM and the dependencies are correctly being tracked there.